File: go/xor_shift_gen/xor.go
package main

import "fmt"

func main() {
    const SEED uint32 = 1
    var val uint32 = SEED
    var period int64 = 0
    var keepgoing bool = true

    for keepgoing {
        val ^= val << 13
        val ^= val >> 17
        val ^= val << 5
        if (val & 0xffffff00) == 0 {
            fmt.Printf("period %d, val %x\n", period, val)
        }
        period++
        if val == SEED {
            keepgoing = false
        }
    }

    fmt.Printf("%d\n", period)

}

File: python/xor_shift_gen/xor.py

def main():
    SEED = 1;
    val = SEED;
    period = 0;
    keepgoing = True

    while (keepgoing):
        val ^= (val << 13) & 0xffffffff;
        val ^= val >> 17;
        val ^= (val << 5) & 0xffffffff;

        if ( (val & 0xffffff00) == 0):
            print('period {}, val {:x}'.format(period, val));

        period += 1;
        if (val == SEED):
            keepgoing = False

    print('{}'.format(period));

if __name__ == "__main__":
    main()